testrunner/src/mill/testrunner/TestRunnerUtils.scala

```diff
@@ -208,6 +210,145 @@ import scala.jdk.CollectionConverters.IteratorHasAsScala
     (doneMessage, results.toSeq)
   }
 
+  def stealTasks(
+      testClasses: Seq[(Class[?], Fingerprint)],
+      testReporter: TestReporter,
+      runner: Runner,
+      stealFolder: os.Path,
+      selectorFolder: os.Path
+  )(implicit ctx: Ctx.Log): (String, Iterator[TestResult]) = {
+    val events = new ConcurrentLinkedQueue[Event]()
+    val doneMessage = {
+
+      val taskQueue = mutable.Queue.empty[Task]
+
+      @tailrec
+      def stealFromSelectorFolder(): Boolean = {
```

Comment:
Can we move `stealFromSelectorFolder` out of `stealTasks` into its own top-level method and make it return a `Option[String]` with the successfully stolen class name? That will help encapsulate things a bit more and keep the code easier to understand